### ISSUE TITLE
Persistence

### ISSUE THREAD

**Issue by TinaTiel** [TinaTiel/obs-chatbot#73]
There's enough of core available now that it can fully run locally in a headless mode (no GUI) with manually configured commands, settings, etc. 

So, it's time to take the next step and remove the manual configuration / replace it with a persistence layer.
- Save commands
- Save authorized clients (Twitch auth)
- Save client settings
- Dev vs default/prod DB profiles (in-memory with drop-create, vs. from-file with create-only or verify)

**Comment by TinaTiel** [comment]
AFter a verrry long weekend, I've finally put the core persistence mechanisms in place for commands and actions; it is now possible to save commands, pull their definitions from the db, and execute them when requested in chat.

Next steps are to add some caching and persisting the rest of the settings + authorized clients (e.g. refresh tokens).

**Comment by TinaTiel** [comment]
Just added the OAuth persistence! Surprisingly it went in without much fuss...I now can save (refresh) tokens to the DB so re-authorization doesn't need to be performed on every startup. Nifty!

**Comment by TinaTiel** [comment]
Need to take a step back and add ownership to each object. For now, it'll all be the same system owner, but it needs to be able to support arbitrary owners. I'll push back those ownership calls as far away from the core as possible, ideally to the rest layer if possible (in that case, probably using the session rather than requiring user ids in plaintext over the request)

**Comment by TinaTiel** [comment]
Ownership has been added to the existing work. Intent is to get the owner id at the controller primarily. There is a slightly awkward case atm for getting it when finding a command by name, so I added an OwnerService that it can use to get the implied owner; in theory this could be swapped out for something based in the security context, or possible make the OwnerService itself request scoped, etc. For now, not going to worry about it for desktop apps and just have it return a system user.
